src/Metadata/Extractor/DynamicResourceExtractorInterface.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Metadata\Extractor;
+
+/**
+ * Extracts a dynamic resource (used by GraphQL for nested resources).
+ *
+ * @author Alan Poulain <contact@alanpoulain.eu>
+ */
+interface DynamicResourceExtractorInterface extends ResourceExtractorInterface
```

Comment:
do we really need to add a new interface? its wanted that a `ResourceExtractorInterface` doesn't have an `addResource` we're using a Factory to prepare resources inside it no?

Reply:
We need an interface since this method needs to be called from the type builder.